xbmc/video/windows/GUIWindowFullScreen.cpp

```diff
@@ -473,3 +474,11 @@ bool CGUIWindowFullScreen::HasVisibleControls()
 {
   return m_controlStats->nCountVisible > 0;
 }
+
+int CGUIWindowFullScreen::GetOSDWindowID()
```

Comment:
Agreed, this is hacky and a new class is needed. I left out this change because the PR was already massive. I'll try refactoring this, but if it touches too many files then it'll warrant its own PR.